### PR TITLE
Fix device mountable volume names in DSW

### PR DESCRIPTION
**Pull request by cofyc**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>
> /kind api-change

/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

This helps to serialize actions on local volume global path. This prevents races on same local volume, see https://github.com/kubernetes/kubernetes/issues/71438#issuecomment-442435242.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Partially address #71438

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Fix device mountable volume names in DSW to prevent races in device mountable plugin, e.g. local.
```

### CODE REVIEW

File: pkg/volume/local/local.go [hunk 1]
```diff
@@ -16,6 +16,11 @@ limitations under the License.
 
 package local
 
+//
```
Comment (resouer) [inline pkg/volume/local/local.go]: I would suggest move this comment to where the lock is used below. Nobody will check notes here I'm afraid.

Comment (cofyc) [reply pkg/volume/local/local.go]: Good suggestion! Done.

**Review by resouer — COMMENTED** [review]
Over looks good :-)

**Comment by msau42** [comment]
UnmountDevice is not supposed to be called until all references are gone.  And should be [serialized with MountVolume](https://sourcegraph.com/github.com/kubernetes/kubernetes/-/blob/pkg/volume/util/operationexecutor/operation_executor.go#L745).  I think we should investigate potential issues with operation executor because it's expected to be able to handle this race.

**Comment by cofyc** [comment]
Oh, if it's expected to handle this race, I am going to check it.

**Comment by cofyc** [comment]
@msau42 

PR has been updated. In testing, I found DSW generates different volume names for same volume, so even if operation executor uses `EmptyUniquePodName` as operation pod name, operations are not serialized.

**Comment by cofyc** [comment]
`util.GetUniqueVolumeNameForNonAttachableVolume` name is a bit confusing now.  Give it a new name and update related code in https://github.com/kubernetes/kubernetes/pull/71509/commits/7a093f233f80c3f6eaf8af331ea0be099e7bb17d.

cc @saad-ali

**Comment by msau42** [comment]
This lgtm.

/assign @jingxu97 
for final review

**Comment by jingxu97** [comment]
@cofyc Thanks a lot for the fix! Could you add a test in desired-state_of_world to verify the generated volume name for different type of plugins? For example, if the device mountable volume is used by two or more pods, they should generate the same unique volume names instead of multiple ones? 
Please let me know if you have questions.

**Comment by cofyc** [comment]
@jingxu97 

Sure, done in https://github.com/kubernetes/kubernetes/pull/71509/commits/0f9e37f2e8050cbcf6590e2c42b8ba2b317006f6.

 I'm not sure if there is another way, testing.FakeVolumePlugin implements all volume interfaces, to mock device mountable, attachable and basic volume plugins, I add some fake volume plugin implementations in test file.

**Review by saad-ali — APPROVED** [review]
/lgtm
/approve

**Pull request merged**